sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java

```diff
@@ -38,57 +38,69 @@ public final class LoggingX509ExtendedTrustManager extends DelegatingX509Extende
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, null, null);
-        super.checkServerTrusted(chain, authType);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType), CounterParty.SERVER, chain, authType, null, null);
     }
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType, Socket socket) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, socket, null);
-        super.checkServerTrusted(chain, authType, socket);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType, socket), CounterParty.SERVER, chain, authType, socket, null);
     }
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType, SSLEngine sslEngine) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, null, sslEngine);
-        super.checkServerTrusted(chain, authType, sslEngine);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType, sslEngine), CounterParty.SERVER, chain, authType, null, sslEngine);
     }
 
-    private static void logCertificate(CounterParty counterParty, X509Certificate[] chain, String authType, Socket socket, SSLEngine sslEngine) {
+    private static void checkTrusted(TrustManagerRunnable runnable,
+                                     CounterParty counterParty,
+                                     X509Certificate[] chain,
+                                     String authType,
+                                     Socket socket,
+                                     SSLEngine sslEngine) throws CertificateException {
+
         String certificateChain = Arrays.toString(chain);
 
         Optional<String> classNameLogMessage = getClassnameOfEitherOrOther(socket, sslEngine)
-                .map(className -> ", while also using the " + className + "");
+                .map(className -> ", while also using the " + className);
         Optional<String> hostAndPortLogMessage = getHostAndPortOfEitherOrOther(socket, sslEngine)
                 .map(hostAndPort -> "[" + hostAndPort + "]");
 
         String logMessage = String.format(LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""), counterParty, certificateChain);
         LOGGER.debug(logMessage);
+
+        try {
+            runnable.run();
+            String okMessage = String.format(VALIDATION_PASSED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            LOGGER.debug(okMessage);
+        } catch (CertificateException e) {
+            String nokMessage = String.format(VALIDATION_FAILED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            LOGGER.debug(nokMessage, e);
```

Comment:
can you make it possible for users to enable logging of certificate exception but not successful check?

some possibilities:
1. different loggers:
```
Logger SUCCESS_LOGGER = getLogger(LoggingX509ExtendedTrustManager.class.getName() + ".success");
Logger SUCCESS_LOGGER = getLogger(LoggingX509ExtendedTrustManager.class.getName() + ".exception");
```

2. slf4j markers

3. allow slf4j level to be passed:
```
TrustManagerUtils.createLoggingTrustManager(delegate,
    org.slf4j.event.Level.DEBUG, // level for success
    org.slf4j.event.Level.WARN); // level for exception
```

4. protected methods for subclass overriding

Reply:
I think the first option would be the better one. In that way you can disable the part which you don't want in your logging configuration file